sklearn/decomposition/_nmf.py

```diff
@@ -1434,3 +1607,358 @@ def inverse_transform(self, W):
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Data matrix to be decomposed
+
+        y : Ignored
+
+        W : array-like of shape (n_samples, n_components)
+            If init='custom', it is used as initial guess for the solution.
+
+        H : array-like of shape (n_components, n_features)
+            If init='custom', it is used as initial guess for the solution.
+            If update_H=False, it is used as a constant, to solve for W only.
+
+        update_H : bool, default=True
+            If True, both W and H will be estimated from initial guesses,
+            this corresponds to a call to the 'fit_transform' method.
+            If False, only W will be estimated, this corresponds to a call
+            to the 'transform' method.
+
+        Returns
+        -------
+        W : ndarray of shape (n_samples, n_components)
+            Transformed data.
+
+        H : ndarray of shape (n_components, n_features)
+            Factorization matrix, sometimes called 'dictionary'.
+
+        n_iter_ : int
+            Actual number of iterations.
+
+        iter_offset : int, default=0
+            Number of previous iterations completed used for
+            initialization, only used in
+            :class:`sklearn.decomposition.MiniBatchNMF`.
+
+        A : array-like of shape (n_components, n_features)
+            Initial guess for the numerator auxiliary function
+
+        B : array-like of shape (n_components, n_features)
+            Initial guess for the denominator auxiliary function
+        """
+        check_non_negative(X, "NMF (input X)")
+        # check parameters
+        self._check_params(X)
+
+        if X.min() == 0 and self._beta_loss <= 0:
+            raise ValueError("When beta_loss <= 0 and X contains zeros, "
+                             "the solver may diverge. Please add small values "
+                             "to X, or use a positive beta_loss.")
+
+        n_samples, n_features = X.shape
+        # initialize or check W and H
+        W, H = self._check_w_h(X, W, H, update_H)
+
+        l1_reg_W, l1_reg_H, l2_reg_W, l2_reg_H = _compute_regularization(
+            self.alpha, self.l1_ratio, self.regularization)
+
+        # Initialize auxiliary matrices
+        A = H.copy()
+        B = np.ones(H.shape, dtype=H.dtype)
+
+        if self.solver == 'mu':
+            W, H, n_iter, iter_offset, A, B = _fit_multiplicative_update(
+                X, W, H, A, B, self._beta_loss, self._batch_size, 0,
+                self.max_iter, self.tol,
+                l1_reg_W, l1_reg_H, l2_reg_W, l2_reg_H,
+                update_H, self.verbose, self.forget_factor)
+        else:
+            raise ValueError("Invalid solver parameter '%s'." % self.solver)
+
+        return W, H, n_iter, iter_offset, A, B
+
+    def partial_fit(self, X, y=None, **params):
+        has_components = not hasattr(self, 'components_')
+
+        if not has_components:
```

Comment:
well I meant `has_components = hasattr(self, 'components_')`, otherwise it's quite confusing :)